pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -249,8 +249,8 @@ def _evaluate(self, model: LightningModule, dataloaders, max_batches: int, test_
             dl_outputs = []
 
             # on TPU we have to wrap it under the ParallelLoader
-            if self.use_tpu and self.tpu_id is None:
-                device = xm.xla_device()
+            if self.use_tpu:
+                device = xm.xla_device(self.tpu_id) if self.tpu_id is not None else xm.xla_device()
```

Comment:
According to docs, the None case does not need to be handled explicitly:
https://pytorch.org/xla/release/1.5/index.html#torch_xla.core.xla_model.xla_device
`device = xm.xla_device(self.tpu_id)` is fine even if `self.tpu_id = None`

Reply:
@awaelchli Yea, it does work without it too. We could refactor these instances across the project.

Reply:
I have refactored it here #1756 so if you do it here too, then I think it's pretty much all.

Reply:
Does it also work for older versions... =)

Reply:
@Borda Do you mean older versions of lightning? I've noticed `xm.xla_device(None)` working earlier too. The docs @awaelchli pointed out though, confirms that it should work well.

Reply:
I was thinking of older version of XLA but there is no reason why it should not :]